src/cmd/update.rs

```diff
@@ -0,0 +1,68 @@
+use std::fs;
+use std::fs::File;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+
+use anyhow::Result;
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const RELEASES_LIST: &str = "https://api.github.com/repos/sbstp/kubie/contents/releases/linux?ref=master";
+const RELEASE_BASE_URL: &str = "https://github.com/sbstp/kubie/raw/master/releases/linux/amd64";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+struct TreeUrl {
+    git_url: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Tree {
+    tree: Vec<KubieVersion>,
+}
+
+#[derive(Debug, Deserialize)]
+struct KubieVersion {
+    path: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_version = get_latest_version()?;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let resp = attohttpc::get(format!("{}/{}/{}", RELEASE_BASE_URL, latest_version, FILENAME)).send()?;
+        if resp.is_success() {
+            let file = File::create(FILENAME)?;
+            let _n = resp.write_to(file)?;
+            let old_file = which::which(FILENAME).unwrap();
+            let res = replace_file(old_file.to_string_lossy().to_string(), FILENAME.to_string());
+            if res.is_ok() {
+                println!("Kubie has been updated successfully. Enjoy :)");
+            } else {
+                println!(
+                    "The update failed : ({}), maybe consider using sudo ?",
+                    res.err().unwrap()
+                );
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn get_latest_version() -> Result<String> {
+    let tree_url: Vec<TreeUrl> = attohttpc::get(&RELEASES_LIST).send()?.json()?;
+    let tree: Tree = attohttpc::get(&tree_url[0].git_url).send()?.json()?;
+    let latest_version = &tree.tree.last().unwrap().path;
+    Ok(latest_version.to_string())
+}
+
+pub fn replace_file(old_file: String, new_file: String) -> std::io::Result<()> {
```

Comment:
Instead of using an owned `String` here, you should use a reference because there's no need for this function to own anything. Also the `old_file` is a `PathBuf` which is similar to `String` but `PathBuf` allows non unicode characters.

So this method could be `replace_file(old_file: &PathBuf, new_file: &str) ...`, simply sending a reference to the `PathBuf`.

However, if you read the documentation of the [`std::path`](https://doc.rust-lang.org/stable/std/path/index.html) module, you will see that you should use `&Path` instead of `&PathBuf` because `PathBuf` can deref into `Path`.

So the final code should look like `replace_file(old_file: &Path, new_file: &str) ...`.

Reply:
Thanks for the review.
The actual method look like this now : 
`pub fn replace_file(old_file: &Path, new_file: &Path) -> std::io::Result<()>`